sci-libs/oce/oce-0.18.3.ebuild

```diff
@@ -0,0 +1,77 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit cmake-utils eutils check-reqs multilib java-pkg-opt-2
+
+DESCRIPTION="Development platform for CAD/CAE, 3D surface/solid modeling and data exchange"
+HOMEPAGE="https://github.com/tpaviot/oce"
+SRC_URI="https://github.com/tpaviot/oce/archive/OCE-$PV.tar.gz"
+
+LICENSE="|| ( Open-CASCADE-LGPL-2.1-Exception-1.0 LGPL-2.1 )"
+SLOT="${PV}"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="examples freeimage gl2ps openmp +tbb +vtk X"
+REQUIRED_USE="?? ( openmp tbb )"
+
+MY_VTK="vtk-8.1"
+MY_P="oce-OCE-${PV}"
+S="${WORKDIR}/${MY_P}"
+DEPEND="
+	dev-lang/tcl:0=
+	dev-lang/tk:0=
+	dev-tcltk/itcl
+	dev-tcltk/itk
+	dev-tcltk/tix
+	media-libs/ftgl
+	media-libs/freetype
+	virtual/glu
+	virtual/opengl
+	x11-libs/libXmu
+	freeimage? ( media-libs/freeimage )
+	gl2ps? ( x11-libs/gl2ps )
+	tbb? ( dev-cpp/tbb )
+	vtk? ( =sci-libs/${MY_VTK}*[boost,imaging,qt5,python,rendering,views,xdmf2] )"
+RDEPEND="${DEPEND}"
+
+CHECKREQS_MEMORY="256M"
+CHECKREQS_DISK_BUILD="3584M"
+
+pkg_setup() {
+	check-reqs_pkg_setup
+}
+
+src_prepare() {
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	# From BUILD.Unix.md
+	local mycmakeargs=(
+		-DOCE_INSTALL_PREFIX="/usr"
+		-DOCE_ENABLE_DEB_FLAG=off
+		-DOCE_COPY_HEADERS_BUILD=yes
+		-DOCE_DRAW=$(usex X)
```

Comment:
Should the effect of the `X` USE flag be a bit wider than `OCE_DRAW`? Shouldn't it control something like `OCE_DISABLE_X11` and allow use to skip X-related dependencies (`x11-libs/libXmu` and Qt)?

As it is, disabling a "test harness" (the description for the `OCE_DRAW` flag) seems rather inconsequential to me and maybe that the `X` USE flag should be removed altogether.

Reply:
Since this is a visualization library I'd say let's just drop the X and move -DOCE_DRAW to be set to enabled enabled by default. The binary is small (14k) and seems to be useful for anyone developing with OCE.